### PR TITLE
refactor(money-input): use hooks

### PR DESCRIPTION
**Pull request by montezume**
#### Summary

Refactors the `MoneyInput` component to use react hooks.

### CODE REVIEW

File: src/components/inputs/money-input/money-input.js [hunk 1]
```diff
@@ -502,202 +336,369 @@ class MoneyInput extends React.Component {
...
+        props.onChange({
+          persist: () => {},
+          target: {
+            id: MoneyInput.getAmountInputId(props.id),
+            name: getAmountInputName(props.name),
+            value: event.target.value,
+          },
+        });
+      }
+    },
```
Comment (tdeekens) [inline src/components/inputs/money-input/money-input.js]: I feel so stupid for still not getting why onChange aint a dep but some part of me gets it.

File: src/components/inputs/money-input/money-input.js [hunk 2]
```diff
@@ -502,202 +336,369 @@ class MoneyInput extends React.Component {
...
+            isDisabled={props.isDisabled || props.isReadOnly}
+            isSearchable={false}
+            components={{
+              // eslint-disable-next-line react/display-name
+              SingleValue: innerProps => (
+                <SingleValue {...innerProps} id={id} />
+              ),
+              DropdownIndicator,
+            }}
+            options={options}
```
Comment (tdeekens) [inline src/components/inputs/money-input/money-input.js]: Hm.

File: src/components/inputs/money-input/money-input.js [hunk 3]
```diff
@@ -502,202 +336,369 @@ class MoneyInput extends React.Component {
...
+            name: getAmountInputName(props.name),
             value: formattedAmount,
           },
         };
-        this.props.onChange(fakeEvent);
+        props.onChange(fakeEvent);
       }
     }
-  };
+  }, [props.id, props.name, props.value.amount]);
```
Comment (islam3zzat) [inline src/components/inputs/money-input/money-input.js]: should we include the `props.value.currencyCode`?

File: src/components/inputs/money-input/money-input.js [hunk 4]
```diff
@@ -502,202 +336,369 @@ class MoneyInput extends React.Component {
...
+              name: getAmountInputName(props.name),
+              value: nextAmount,
+            },
+          });
+        }
+
+        amountInputRef.current.focus();
+      }
+    },
+    [props.id, props.name]
```
Comment (islam3zzat) [inline src/components/inputs/money-input/money-input.js]: should we add `props.value.currencyCode` and `props.value.amount`?

File: src/components/inputs/money-input/money-input.js (hunk 3 above)
Comment (emmenko) [reply src/components/inputs/money-input/money-input.js]: Yes we should include all the props used within the hook.

PS: do we have the eslint rules for hooks enabled?

File: src/components/inputs/money-input/money-input.js [hunk 5]
```diff
@@ -502,202 +336,369 @@ class MoneyInput extends React.Component {
...
+      <div
+        ref={containerRef}
+        css={css`
+          font-family: inherit;
+          width: 100%;
+          position: relative;
+          display: flex;
+        `}
+        data-testid="money-input-container"
+        onBlur={event => {
```
Comment (emmenko) [inline src/components/inputs/money-input/money-input.js]: Nitpick: our this event handler into a `useCallback`

File: src/components/inputs/money-input/money-input.js [hunk 6]
```diff
@@ -502,202 +336,369 @@ class MoneyInput extends React.Component {
...
+              // eslint-disable-next-line react/display-name
+              SingleValue: innerProps => (
+                <SingleValue {...innerProps} id={id} />
+              ),
+              DropdownIndicator,
+            }}
+            options={options}
+            placeholder=""
+            styles={currencySelectStyles}
+            onFocus={() => {
```
Comment (emmenko) [inline src/components/inputs/money-input/money-input.js]: Same here

File: src/components/inputs/money-input/money-input.js [hunk 7]
```diff
@@ -309,190 +309,25 @@ const getAmountInputName = name => (name ? `${name}.amount` : undefined);
...
-          value: event.target.value,
-        },
-      });
-    }
-  };
-
-  handleAmountBlur = () => {
-    const amount = this.props.value.amount.trim();
-    this.setState({ amountHasFocus: false });
+  const { onChange } = props;
```
Comment (emmenko) [inline src/components/inputs/money-input/money-input.js]: Same drill here again

File: src/components/inputs/money-input/money-input.js [hunk 8]
```diff
@@ -502,202 +337,383 @@ class MoneyInput extends React.Component {
...
     }
-  };
+  }, [
+    intl.locale,
+    onChange,
+    props.id,
+    props.name,
+    props.value.amount,
+    props.value.currencyCode,
+  ]);
```
Comment (emmenko) [inline src/components/inputs/money-input/money-input.js]: All required hooks deps 👆

File: src/components/inputs/money-input/money-input.js [hunk 9]
```diff
@@ -502,202 +337,383 @@ class MoneyInput extends React.Component {
...
+          persist: () => {},
+          target: {
+            id: MoneyInput.getAmountInputId(props.id),
+            name: getAmountInputName(props.name),
+            value: event.target.value,
+          },
+        });
+      }
+    },
+    [onChange, props.id, props.name]
```
Comment (emmenko) [inline src/components/inputs/money-input/money-input.js]: All required hooks deps 👆

File: src/components/inputs/money-input/money-input.js [hunk 10]
```diff
@@ -502,202 +337,383 @@ class MoneyInput extends React.Component {
...
+      }
+    },
+    [
+      intl.locale,
+      onChange,
+      props.id,
+      props.name,
+      props.value.amount,
+      props.value.currencyCode,
+    ]
```
Comment (emmenko) [inline src/components/inputs/money-input/money-input.js]: All required hooks deps 👆

**Comment by emmenko** [comment]
The VRTs now render the `menuPortalZIndex` prop in the description...not sure why.

image

**Comment by emmenko** [comment]
FYI: the only thing blocking this is approving the VRTs. Let's wait for @montezume maybe regarding that.

**Comment by montezume** [comment]
@emmenko it does this because we removed a wrapper... that's why we opened https://github.com/commercetools/ui-kit/issues/875

**Comment by emmenko** [comment]
Ah ok, I missed that issue I think. Thanks!

**Pull request merged**